src/hotspot/share/runtime/os.cpp

```diff
@@ -1211,6 +1210,23 @@ bool os::is_first_C_frame(frame* fr) {
   if (old_fp - ufp > 64 * K) return true;
 
   return false;
+#endif
+}
+
+// Looks like all platforms can use the same function to check if C
+// stack is walkable beyond current frame.
+// This version of the method adds more checks that need the thread
+bool os::is_first_C_frame(frame* fr, Thread *t) {
+#ifdef _WINDOWS
+  return true; // native stack isn't walkable on windows this way.
+#else
+  return !fr->can_access_link(t) || os::is_first_C_frame(fr) ||
```

Comment:
Check t for NULL.

Reply:
I would assert for not NULL and ensure the caller only uses this with a non-NULL thread.